Description: Modify the system default
 While the upstream default "auto" is believed to work best on Unity,
 MATE and Ubuntu GNOME, "cjkv" makes sure on other desktops that IBus
 or fcitx is not started by default for all users.
Bug-Ubuntu: https://launchpad.net/bugs/1481025, https://launchpad.net/bugs/1550325, https://launchpad.net/bugs/1551283
Author: Gunnar Hjalmarsson <gunnarhj@ubuntu.com>
Forwarded: not-needed

--- a/default/im-config
+++ b/default/im-config
@@ -1,9 +1,12 @@
 # Default im-config mode (see im-config(8))
 
-# Always start best input method
-IM_CONFIG_DEFAULT_MODE=auto
-# Start best input method only if CJKV environment
-#IM_CONFIG_DEFAULT_MODE=cjkv
+if [ "$XDG_CURRENT_DESKTOP" = 'Unity' -o "$XDG_CURRENT_DESKTOP" = 'MATE' -o "${XDG_CURRENT_DESKTOP##*:}" = 'GNOME' ]; then
+    # Start best input method unless overridden below
+    IM_CONFIG_DEFAULT_MODE=auto
+else
+    # Start best input method only if CJKV environment and not overridden below
+    IM_CONFIG_DEFAULT_MODE=cjkv
+fi
 
 # Set locale dependent preferred IM over standard auto mode
 if [ "${XDG_CURRENT_DESKTOP##*:}" != 'GNOME' -a "${XDG_CURRENT_DESKTOP%%:*}" != 'GNOME-Flashback' ]; then
